### PR TITLE
Rename AccountsDb plugins to Geyser plugins

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
The name "AccountsDb Plugin" isn't accurate for the implementation, since plugins can now stream blocks and transactions in addition to accounts.

#### Summary of Changes
Rename to the generic term `geyser`
Currently doc references to the `solana-accountsdb-plugin-postgres` repository have been left unchanged, to be updated when that repo is renamed

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
Rebasing due to toml sort changes just merged, but passed CI previously

File: docs/src/developing/plugins/geyser-plugins.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-consume data from these external data stores with the possibility of more
-flexible and targeted optimizations such as caching and indexing. This allows
-the validator to focus on processing transactions without being slowed down by
-busy RPC requests.
+accounts, slots, blocks, and transactions can be transmitted to external data
+stores such as relational databases, NoSQL databases or Kafka. RPC services then
+can be developed to consume data from these external data stores with the
+possibility of more flexible and targeted optimizations such as caching and
+indexing. This allows the validator to focus on processing transactions without
+being slowed down by busy RPC requests.
```
Comment (lijunwangs) [inline docs/src/developing/plugins/geyser-plugins.md:15]: Maybe we should update it here with the Geyer plugin name. For example 'support a plugin mechanism named Geyer Plugins'..

File: validator/src/bin/solana-test-validator.rs [hunk 2]
```diff
@@ -312,12 +312,13 @@ fn main() {
                 ),
         )
         .arg(
-            Arg::with_name("accountsdb_plugin_config")
-                .long("accountsdb-plugin-config")
+            Arg::with_name("geyser_plugin_config")
+                .long("geyser-plugin-config")
+                .alias("accountsdb-plugin-config")
```
Comment (lijunwangs) [inline validator/src/bin/solana-test-validator.rs:317]: backward compatibility?

**Review by lijunwangs — COMMENTED** [review]
looks good -- just  a couple of nits

File: Cargo.toml [hunk 3]
```diff
@@ -1,10 +1,9 @@
 [workspace]
 members = [
-    "account-decoder",
```
[the remark this answers is not in the document]
Comment (CriesofCarrots) [reply Cargo.toml:3]: Ah, shoot. Yeah, that's a bad rebase on https://github.com/solana-labs/solana/pull/23602
Will fixup

File: validator/src/bin/solana-test-validator.rs (hunk 2 above)
Comment (CriesofCarrots) [reply validator/src/bin/solana-test-validator.rs:317]: Yep, I figured we might as well avoid breaking the validator interface, even though I guess users will have to redeploy on that update anyway.

**Review by lijunwangs — DISMISSED** [review]
Looks good. Thanks!

**Review by t-nelson — APPROVED** [review]
Changes here lgtm!  The rollout is the main open question.

Tyera and I discussed at lunch and it seems least intrusive if we can release the last version of the `solana-accountsdb-plugin-interface` and the first version of the `solana-geyser-plugin-interface` with an otherwise unchanged API/ABI/behavior set. So the steps are basically...
1. land imminent additions to `solana-accountsdb-plugin-interface`
2. publish crate (in-band?)
3. merge this PR
4. publish crate (out-of-band?)
5. (opt?) publish a final out-of-band `solana-accountsdb-plugin-interface` crate with all public symbols marked deprecated

how does this sound to you @lijunwangs?

**Comment by CriesofCarrots** [comment]
I'm gonna try the auto-backport. We'll see how it goes

**Pull request merged**